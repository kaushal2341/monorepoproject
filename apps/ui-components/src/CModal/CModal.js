import React from 'react';
import Modal from 'react-bootstrap/Modal';
import PropTypes from 'prop-types';

const CModal = props => {
    const {
        animation,
        asBody,
        asTitle,
        autoFocus,
        backdrop,
        backdropClassName,
        bodyChildren,
        bsPrefix,
        bsPrefixBody,
        bsPrefixFooter,
        bsPrefixHeader,
        bsPrefixTitle,
        centered,
        closeButton,
        closeLabel,
        container,
        dialogAs,
        dialogClassName,
        enforceFocus,
        footerAs,
        footerChildren,
        keyboard,
        manager,
        modalHeading,
        onEnter,
        onEntered,
        onEntering,
        onEscapeKeyDown,
        onExit,
        onExited,
        onExiting,
        onHide,
        onShow,
        restoreFocus,
        scrollable,
        show,
        size
    } = props;
    return (
        <Modal
            animation={animation}
            autoFocus={autoFocus}
            backdrop={backdrop}
            backdropClassName={backdropClassName}
            bsPrefix={bsPrefix}
            centered={centered}
            container={container}
            dialogAs={dialogAs}
            dialogClassName={dialogClassName}
            enforceFocus={enforceFocus}
            id="modal"
            keyboard={keyboard}
            manager={manager}
            onEnter={onEnter}
            onEntered={onEntered}
            onEntering={onEntering}
            onEscapeKeyDown={onEscapeKeyDown}
            onExit={onExit}
            onExited={onExited}
            onExiting={onExiting}
            onHide={onHide}
            onShow={onShow}
            restoreFocus={restoreFocus}
            scrollable={scrollable}
            show={show}
            size={size}>

            <Modal.Header
                id="modal-header"
                closeButton={closeButton}
                closeLabel={closeLabel}
                bsPrefix={bsPrefixHeader}>
                <Modal.Title
                    id="title"
                    as={asTitle}
                    bsPrefix={bsPrefixTitle}>
                    {modalHeading}
                </Modal.Title>
            </Modal.Header>

            <Modal.Body
                id="modal-body"
                as={asBody}
                bsPrefix={bsPrefixBody}
                children={bodyChildren}
            >
            </Modal.Body>
            <Modal.Footer
                id="modal-footer"
                as={footerAs}
                bsPrefix={bsPrefixFooter}
                children={footerChildren}
            >
            </Modal.Footer>

        </Modal>
    );
};

CModal.defaultProps = {
    centered: true,
    closeButton: true,
    onHide: () => {}
};

CModal.propTypes = {
    animation: PropTypes.bool,
    asBody: PropTypes.elementType, // as
    asTitle: PropTypes.elementType, // as
    autoFocus: PropTypes.bool,
    backdrop: PropTypes.bool || PropTypes.string,
    backdropClassName: PropTypes.string,
    bsPrefix: PropTypes.string,
    bsPrefixBody: PropTypes.string,
    bsPrefixFooter: PropTypes.string,
    bsPrefixHeader: PropTypes.string,
    bsPrefixTitle: PropTypes.string,
    centered: PropTypes.bool,
    closeButton: PropTypes.bool,
    closeLabel: PropTypes.string,
    container: PropTypes.any,
    dialogAs: PropTypes.elementType,
    dialogClassName: PropTypes.string,
    enforceFocus: PropTypes.bool,
    footerAs: PropTypes.elementType,
    id: PropTypes.string,
    keyboard: PropTypes.bool,
    manager: PropTypes.object,
    modalHeading: PropTypes.string.isRequired,
    onEnter: PropTypes.func,
    onEntered: PropTypes.func,
    onEntering: PropTypes.func,
    onEscapeKeyDown: PropTypes.func,
    onExit: PropTypes.func,
    onExited: PropTypes.func,
    onExiting: PropTypes.func,
    onHide: PropTypes.func.isRequired,
    onShow: PropTypes.func,
    restoreFocus: PropTypes.bool,
    scrollable: PropTypes.bool,
    show: PropTypes.bool,
    size: PropTypes.string,//sm lg xl
};

export default CModal;