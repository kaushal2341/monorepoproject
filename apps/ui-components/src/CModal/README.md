#CModal

`CModal` is a modal component dynamically build using react-bootstrap component.

Props of CModal:

```text
Name	            Type	                Default	    Description
animation	        boolean	                true	    Open and close the Modal with a slide and fade animation.
autoFocus	        boolean	                true	    When true The modal will automatically shift focus to itself when it opens, and replace it to the last focused element when it closes. Generally this should never be set to false as it makes the Modal less accessible to assistive technologies, like screen-readers.
backdrop	        'static' | true         true	    Include a backdrop component. Specify 'static' for a backdrop that doesn't trigger an "onHide" when clicked.
                    | false	
backdropClassName	string		                        Add an optional extra class name to .modal-backdrop It could end up looking like class="modal-backdrop foo-modal-backdrop in".
centered	        boolean		                        vertically center the Dialog in the window
container	        any		
dialogAs	        elementType	           <ModalDialog> A Component type that provides the modal content Markup. This is a useful prop when you want to use your own styles and markup to create a custom modal component.
dialogClassName	    string		                         A css class to apply to the Modal dialog DOM node.
enforceFocus	    boolean	                true	     When true The modal will prevent focus from leaving the Modal while open. Consider leaving the default value here, as it is necessary to make the Modal work well with assistive technologies, such as screen readers.
keyboard	        boolean	                true	     Close the modal when escape key is pressed
manager	            object	        new BootstrapModalManager()	A ModalManager instance used to track and manage the state of open Modals. Useful when customizing how modals interact within a container
onEnter 	        function		                       Callback fired before the Modal transitions in
onEntered	        function		                       Callback fired after the Modal finishes transitioning in
onEntering	        function		                       Callback fired as the Modal begins to transition in
onEscapeKeyDown	    function		                       A callback fired when the escape key, if specified in keyboard, is pressed.
onExit	            function		                       Callback fired right before the Modal transitions out
onExited	        function		                       Callback fired after the Modal finishes transitioning out
onExiting	        function		                       Callback fired as the Modal begins to transition out
onHide	            function		                       A callback fired when the header closeButton or non-static backdrop is clicked. Required if either are specified.
onShow	            function		                       A callback fired when the Modal is opening.
restoreFocus	    boolean	                true	       When true The modal will restore focus to previously focused element once modal is hidden
scrollable	        boolean		                           Allows scrolling the <Modal.Body> instead of the entire Modal when overflowing.
show	            boolean	                false	       When true The modal will show itself.
size	            'sm' | 'lg','xl'		               Render a large, extra large or small modal.
bsPrefix	        string	                'modal'	       Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatchfor working with heavily customized bootstrap css.
closeButton	      boolean	                false	       Specify whether the Component should contain a close button
onHide	            function		                       A Callback fired when the close button is clicked. If used directly inside a Modal component, the onHide will automatically be propagated up to the parent Modal onHide.
as	            elementType	        <DivStyledAsH4>    You can use a custom element type for this component.
                                                or <div>

```

To use this component, create a boolean property, initially set to false  in the state of the component where modal is required.

```javascript

class ModalTestComponent extends PureComponent {

    state = {
        showModal: false
    };

    setShowModal = () => {
        this.setState({showModal: !this.state.showModal});
    };
}
```
Usually modals are opened on certain events or button click.
Pass this `setShowModal` or function that toggles the value of boolean property of state(`showModal`) 
to the eventHandler(here, `onClickHandler`).

```javascript
 render() {
        let footerButton = <CButton id="sabu" name="Sabu"/>;
        let body = <> <h1>This is Modal body</h1><CButton id="sauravi" name="sauravi"/></>;
        return (
            <>
                <CButton id="modal" name="Open Modal" onClickHandler={this.setShowModal}/>
                <CModal show={this.state.showModal}
                        modalHeading="Modal Smriti"
                        size="md"
                        bodyChildren={body}
                        footerChildren={footerButton}
                        onHide={this.setShowModal}
                />
            </>
        );
    }
```
`show` ,`modalHeading`,`bodyChildren` and `onHide` props are required. Other props mentioned above can be used according to requirement.