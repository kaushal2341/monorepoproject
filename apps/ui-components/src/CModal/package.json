{
  "name": "@cogent/c-modal",
  "version": "0.0.0",
  "main": "CModal.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
