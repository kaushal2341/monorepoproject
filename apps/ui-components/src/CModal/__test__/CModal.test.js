import CModal from '../CModal';

expect.addSnapshotSerializer(enzymeSerializer);

describe('CModal Tests', () => {
    let sWrapper;
    beforeEach(() => {
        sWrapper = shallow(<CModal modalHeading="Modal Heading Test"/>);
    });

    test('if CModal is defined', () => {
        expect(sWrapper).toBeDefined();
    });

    describe('Modal Tests', () => {
        test('if has one Modal element', () => {
            expect(sWrapper.find('#modal').length).toEqual(1);
        });

        test('if Modal has all required props', () => {
            let propRequired = [
                'animation',
                'autoFocus',
                'backdrop',
                'backdropClassName',
                'bsPrefix',
                'centered',
                'container',
                'dialogAs',
                'dialogClassName',
                'enforceFocus',
                'id',
                'keyboard',
                'manager',
                'onEnter',
                'onEntered',
                'onEntering',
                'onEscapeKeyDown',
                'onExit',
                'onExited',
                'onExiting',
                'onHide',
                'onShow',
                'restoreFocus',
                'scrollable',
                'show',
                'size'
            ];

            let propsForModal = Object.keys(sWrapper.find('#modal').props());
            propRequired.forEach((propAvail, i) => (
                expect(propAvail).toContain(propsForModal[i])
            ));
        });

        test('if centered props has non empty value', () => {
            expect(sWrapper.find('#modal').prop('centered')).not.toEqual('');
        })
    });


    describe('Modal.Header Tests', () => {
        test('if one Modal.Header element is defined', () => {
            expect(sWrapper.find('#modal-header')).toBeDefined();
            expect(sWrapper.find('#modal-header').length).toBe(1);
        });

        test('if Modal.Header contains required props', () => {
            let propRequired = [
                'id',
                'closeButton',
                'closeLabel',
                'bsPrefix'
            ];

            let propsOfModalHeader = Object.keys(sWrapper.find('#modal-header').props());
            propRequired.forEach((propAvail, i) => (
                expect(propAvail).toContain(propsOfModalHeader[i])
            ));
        });

        test('if closeButton props has non empty value', () => {
            expect(sWrapper.find('#modal-header').prop('closeButton')).not.toEqual('');
        });

    });

    describe('Modal.Title Tests', () => {
        test('if one Modal.Title element is defined', () => {
            expect(sWrapper.find('#title')).toBeDefined();
            expect(sWrapper.find('#title').length).toBe(1);
        });

        test('if Modal.title contains required props', () => {
            let propRequired = [
                'id',
                'as',
                'bsPrefix'
            ];
            let propsOfModalTitle = Object.keys(sWrapper.find('#title').props());
            propRequired.forEach((propAvail, i) => (
                expect(propAvail).toContain(propsOfModalTitle[i])
            ));
        });

        test('if Modal has non empty Heading', () => {
            expect(sWrapper.find('#title').text()).not.toBe("");
        });

    });

    describe('Modal.Body Tests', () => {
        test('if one Modal.Body element is defined', () => {
            expect(sWrapper.find('#modal-body')).toBeDefined();
            expect(sWrapper.find('#modal-body').length).toBe(1);
        });

        test('if Modal.Body contains required props', () => {
            let propRequired = [
                'id',
                'as',
                'bsPrefix',
                'children'
            ];

            let propsOfModalBody = Object.keys(sWrapper.find('#modal-body').props());
            propRequired.forEach((propAvail, i) => (
                expect(propAvail).toContain(propsOfModalBody[i])
            ));
        });
    });

    describe('Modal.Footer Tests', () => {
        test('if one Modal.Footer element is defined', () => {
            expect(sWrapper.find('#modal-footer')).toBeDefined();
            expect(sWrapper.find('#modal-footer').length).toBe(1);
        });

        test('if Modal.Footer contains required props', () => {
            let propRequired = [
                'id',
                'as',
                'bsPrefix',
                'children'
            ];

            let propsOfModalBody = Object.keys(sWrapper.find('#modal-footer').props());
            propRequired.forEach((propAvail, i) => (
                expect(propAvail).toContain(propsOfModalBody[i])
            ));
        });
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(sWrapper).toMatchSnapshot();
        })
    });

});