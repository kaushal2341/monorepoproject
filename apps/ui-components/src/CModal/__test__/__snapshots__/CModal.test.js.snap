// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CModal Tests Snapshot Testing should match snapshot 1`] = `
<Bootstrap(Modal)
  centered={true}
  id="modal"
  onHide={[Function]}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
    id="modal-header"
  >
    <ModalTitle
      id="title"
    >
      Modal Heading Test
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    id="modal-body"
  />
  <ModalFooter
    id="modal-footer"
  />
</Bootstrap(Modal)>
`;
